test/e2e/node/node_problem_detector.go

```diff
@@ -360,6 +360,8 @@ func getNpdPodStat(f *framework.Framework, nodeName string) (cpuUsage, rss, work
 		hasNpdPod = true
 		break
 	}
-	framework.ExpectEqual(hasNpdPod, true)
+	if !hasNpdPod {
+		framework.Failf("No node-problem-detector pod is present in %v", summary.Pods)
```

Comment:
how the output will look like on failure, curious how `%v` will format this?

Reply:
When I was not using any formatting directive in the Failf function an error was shown that Failf needs format directive to run. So, I added %v in this the output statement. I think %v will print list of pods present and no npd pod will be present in that list.

Reply:
When I was not using any formatting directive in the Failf function an error was shown that Failf needs format directive to run. So, I added %v in this the output statement. I think %v will print list of pods present and no npd pod will be present in that list.

Reply:
why not `%+q`? Also the structure will be huge, this is why I wonder how it will look like and whether it will be readable at all.

Reply:
Actually I am unable to figure out how to see the output  the function throws if condition follows, can you tell me how to do it.

Reply:
Here's a short test program:
```
package main

import (
	"fmt"

	"k8s.io/kubelet/pkg/apis/stats/v1alpha1"
)

func main() {
	summary := v1alpha1.Summary{}
	for i := 0; i < 10; i++ {
		summary.Pods = append(summary.Pods, v1alpha1.PodStats{
			PodRef: v1alpha1.PodReference{
				Name: fmt.Sprintf("pod-%d", i),
			},
		})
	}
	fmt.Printf("%v", summary.Pods)
}
```

I can't be run in the Go playground (downloading the imports times out), otherwise I would link to that.

It prints:
```
[{{pod-0  } 0001-01-01 00:00:00 +0000 UTC [] <nil> <nil> <nil> [] <nil> <nil>} {{pod-1  } 0001-01-01 00:00:00 +0000 UTC [] <nil> <nil> <nil> [] <nil> <nil>} {{pod-2  } 0001-01-01 00:00:00 +0000 UTC [] <nil> <nil> <nil> [] <nil> <nil>} {{pod-3  } 0001-01-01 00:00:00 +0000 UTC [] <nil> <nil> <nil> [] <nil> <nil>} {{pod-4  } 0001-01-01 00:00:00 +0000 UTC [] <nil> <nil> <nil> [] <nil> <nil>} {{pod-5  } 0001-01-01 00:00:00 +0000 UTC [] <nil> <nil> <nil> [] <nil> <nil>} {{pod-6  } 0001-01-01 00:00:00 +0000 UTC [] <nil> <nil> <nil> [] <nil> <nil>} {{pod-7  } 0001-01-01 00:00:00 +0000 UTC [] <nil> <nil> <nil> [] <nil> <nil>} {{pod-8  } 0001-01-01 00:00:00 +0000 UTC [] <nil> <nil> <nil> [] <nil> <nil>} {{pod-9  } 0001-01-01 00:00:00 +0000 UTC [] <nil> <nil> <nil> [] <nil> <nil>}]
```

It's a bit better with `%+v`:
```
[{PodRef:{Name:pod-0 Namespace: UID:} StartTime:0001-01-01 00:00:00 +0000 UTC Containers:[] CPU:<nil> Memory:<nil> Network:<nil> VolumeStats:[] EphemeralStorage:<nil> ProcessStats:<nil>} {PodRef:{Name:pod-1 Namespace: UID:} StartTime:0001-01-01 00:00:00 +0000 UTC Containers:[] CPU:<nil> Memory:<nil> Network:<nil> VolumeStats:[] EphemeralStorage:<nil> ProcessStats:<nil>} {PodRef:{Name:pod-2 Namespace: UID:} StartTime:0001-01-01 00:00:00 +0000 UTC Containers:[] CPU:<nil> Memory:<nil> Network:<nil> VolumeStats:[] EphemeralStorage:<nil> ProcessStats:<nil>} {PodRef:{Name:pod-3 Namespace: UID:} StartTime:0001-01-01 00:00:00 +0000 UTC Containers:[] CPU:<nil> Memory:<nil> Network:<nil> VolumeStats:[] EphemeralStorage:<nil> ProcessStats:<nil>} {PodRef:{Name:pod-4 Namespace: UID:} StartTime:0001-01-01 00:00:00 +0000 UTC Containers:[] CPU:<nil> Memory:<nil> Network:<nil> VolumeStats:[] EphemeralStorage:<nil> ProcessStats:<nil>} {PodRef:{Name:pod-5 Namespace: UID:} StartTime:0001-01-01 00:00:00 +0000 UTC Containers:[] CPU:<nil> Memory:<nil> Network:<nil> VolumeStats:[] EphemeralStorage:<nil> ProcessStats:<nil>} {PodRef:{Name:pod-6 Namespace: UID:} StartTime:0001-01-01 00:00:00 +0000 UTC Containers:[] CPU:<nil> Memory:<nil> Network:<nil> VolumeStats:[] EphemeralStorage:<nil> ProcessStats:<nil>} {PodRef:{Name:pod-7 Namespace: UID:} StartTime:0001-01-01 00:00:00 +0000 UTC Containers:[] CPU:<nil> Memory:<nil> Network:<nil> VolumeStats:[] EphemeralStorage:<nil> ProcessStats:<nil>} {PodRef:{Name:pod-8 Namespace: UID:} StartTime:0001-01-01 00:00:00 +0000 UTC Containers:[] CPU:<nil> Memory:<nil> Network:<nil> VolumeStats:[] EphemeralStorage:<nil> ProcessStats:<nil>} {PodRef:{Name:pod-9 Namespace: UID:} StartTime:0001-01-01 00:00:00 +0000 UTC Containers:[] CPU:<nil> Memory:<nil> Network:<nil> VolumeStats:[] EphemeralStorage:<nil> ProcessStats:<nil>}]
```

Not particularly readable, but IMHO still better than nothing. I don't think a helper function that extracts just the names is justified.

Reply:
@NikhilSharmaWe can you please change the formatter to `%+v`. @pohly thank you for providing an easy way to validate this change!

Reply:
@SergeyKanzhelev did that already.